File: patterns/design-patterns-structural/src/main/java/com/baeldung/hexagonal/port/StudentService.java
package com.baeldung.hexagonal.port;

import java.util.List;

import com.baeldung.hexagonal.domain.Student;


public interface StudentService {

    List<Student> readAll();
    
    Student read(Long id);
    
    Student create(Student student);
    
    Student update(Long id, Student student);
    
    void delete(Long id);
}

File: jee-kotlin/src/main/kotlin/com/baeldung/jeekotlin/service/StudentService.kt
package com.baeldung.jeekotlin.service

import com.baeldung.jeekotlin.entity.Student
import javax.ejb.Stateless
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

@Stateless
open class StudentService {

    @PersistenceContext
    private lateinit var entityManager: EntityManager

    open fun create(student: Student) = entityManager.persist(student)

    open fun read(id: Long): Student? = entityManager.find(Student::class.java, id)

    open fun update(student: Student) = entityManager.merge(student)

    open fun delete(id: Long) = entityManager.remove(read(id))
}